File: Java/SETPractice/src/main/java/SOLIDPrinciple/LiskovSubstitution/SavingsAccount.java
package SOLIDPrinciple.LiskovSubstitution;

public class SavingsAccount extends BankAccount{
    public double InterestRate;
    public SavingsAccount(String accountNumber, double balance, double interestRate) {
        super(accountNumber, balance);
        InterestRate = interestRate;
    }

    @Override
    public void Withdraw(double amount) {
        if (amount <= Balance)
        {
            Balance -= amount;
            System.out.println("Remaining Balance in your Savings Account is " + Balance);
        }
        else
            System.out.println("Insufficient Balance in your Savings Account");
    }
}

File: DotNet/SETPractice/SOLIDPrinciples/LiskovSubstitution/SavingsAccount.cs
namespace SOLIDPrinciples.LiskovSubstitution
{
    public class SavingsAccount : BankAccount
    {
        public decimal InterestRate { get; set; }

        public SavingsAccount(string accountNumber, decimal balance, decimal interestRate) : base(accountNumber, balance)
        {
            InterestRate = interestRate;
        }

        public override void Withdraw(decimal amount)
        {
            if (amount <= Balance)
            {
                Balance -= amount;
                Console.WriteLine($"Remaining Balance in your Savings Account is {Balance}");
            }
            else
                Console.WriteLine("Insufficient Balance in your Savings Account");
        }
    }
}
